### PR TITLE
i18n(zh-cn): Update `astro-components.md` and `typescript.md`

### PR DESCRIPTION
**Pull request by mogeko**
#### What kind of changes does this PR include?

- New or updated content
- Translated content

#### Description

I updated the Chinese translations of `astro-components.md` and `typescript.md` to catch up with the English version.

`typescript.md` in particular, it is so behind the English version that I almost rewrote it with this update.

For `astro-components.md`, I removed `export` requirement for `Props` #1632.

By the way, `hacktoberfest-accepted` please.  :kissing_heart:

### CODE REVIEW

**Review by delucis — CHANGES_REQUESTED** [review]
Hi @mogeko! Sorry to only get to this now. There have been a few more changes to the English versions, would you be able to update this PR with those?

- astro-components: https://github.com/withastro/docs/commit/b34822eb26f396fe6e6e2c08822122a0cf295766#diff-2cea51d918d3ae4fedfce864624f37591ca58c59140f2f60c55e53641e91a661
- typscript: https://github.com/withastro/docs/commit/1e4d6f86cca3fd2d82dcab14f89566a3dfe14ebf#diff-8bd9acd482c1c173d693abf42692979a226d6c46a9ca5f3980214d168e85a076

**Comment by mogeko** [comment]
Of course! I will update this PR after I finish it.

**Comment by delucis** [comment]
Thank you!

**Comment by mogeko** [comment]
Hi @delucis!

As you can see, I retranslated `astro-components.md` and `typscript.md` according to the latest English version of the document. (`astro-components.md` to #2102, `typscript.md` to #2030)

It seems that there is also an article [`client-side-scripts.md`](src/pages/en/guides/client-side-scripts.md) related to part of `astro-components.md`. There is no Chinese translation yet. But please merge it first. After I prepare the translation of `client-side-scripts.md`, I will submit a new PR.

File: src/pages/zh-cn/guides/typescript.md [hunk 1]
```diff
@@ -79,5 +98,101 @@ const { greeting = 'Hello', name } = Astro.props
...
+
+- 如果你的组件必须将一个子组件传递给默认插槽，你可以使用 `type Props = { children: any; };`。
+
+## 类型工具包
+
+<Since v="1.6.0" />
+
+Astro comes with some built-in utility types for common prop type patterns. These are available under the `astro/types` entrypoint.
+
+Astro 为常见的组件参数的类型模式准备了一些实用的类型工具集。这些可以通过在代码中引入 `astro/types` 来使用。
```
Comment (delucis) [inline src/pages/zh-cn/guides/typescript.md:113]: Looks like the original English text got left in here:

```suggestion
Astro 为常见的组件参数的类型模式准备了一些实用的类型工具集。这些可以通过在代码中引入 `astro/types` 来使用。
```

File: src/pages/zh-cn/guides/typescript.md [hunk 2]
```diff
@@ -79,5 +98,101 @@ const { greeting = 'Hello', name } = Astro.props
...
+
+```astro title="src/components/Link.astro" ins="HTMLAttributes" ins="HTMLAttributes<'a'>"
+---
+type Props = astroHTML.JSX.AnchorHTMLAttributes;
+const { href, ...attrs } = Astro.props;
+---
+<a {href} {...attrs}>
+  <slot />
+</a>
+```
```
Comment (delucis) [inline src/pages/zh-cn/guides/typescript.md:129]: I think this code example is still showing the old version. The new example looks more like this:

````suggestion
```astro title="src/components/Link.astro" ins="HTMLAttributes" ins="HTMLAttributes<'a'>"
---
import { HTMLAttributes } from 'astro/types'
// use a `type`
type Props = HTMLAttributes<'a'>;
// or extend with an `interface`
interface Props extends HTMLAttributes<'a'> {
  myProp?: boolean;
}
const { href, ...attrs } = Astro.props;
---
<a {href} {...attrs}>
  <slot />
</a>
```
````

Would be good to translate the “use a `type`” and “or extend with and `interface`” comments.

File: src/pages/zh-cn/guides/typescript.md [hunk 3]
```diff
@@ -79,5 +98,101 @@ const { greeting = 'Hello', name } = Astro.props
...
+
+### 同时使用多个 JSX 框架所带来的类型错误
+
+在同一个项目中使用多个 JSX 框架时可能会出现问题，因为每个框架在 `tsconfig.json` 中的不同需求有时会相互冲突。
+
+**解决方案**：根据你最常用的框架，将 [`jsxImportSource` 这一设置项](https://www.typescriptlang.org/tsconfig#jsxImportSource)设置为 `react`（默认）、`preact` 或 `solid-js`。然后，在来自不同框架的任何冲突文件中使用[编译指示（pragma comment）](https://www.typescriptlang.org/docs/handbook/jsx.html#configuring-jsx)进行注释。
+
+For the default setting of `jsxImportSource: react`, you would use:
+
+对于默认设置 `jsxImportSource: react`，你可以使用：
```
Comment (delucis) [inline src/pages/zh-cn/guides/typescript.md:182]: Duplicate English sentence here:

```suggestion
对于默认设置 `jsxImportSource: react`，你可以使用：
```

**Review by delucis — COMMENTED** [review]
Thanks @mogeko! Spotted a few small details to clean up and then this should be ready to merge 🎊

**Comment by mogeko** [comment]
@delucis There should be fine now! Please check.

**Review by delucis — APPROVED** [review]
Thanks @mogeko! Excellent contribution ❤️

**Pull request merged**